### PR TITLE
docs(authentication): add MFA example

### LINKED ISSUE #4418: [📚] Add documentation on MFA like authentication flow
## Documentation Feedback

Please describe your documentation issue or suggested improvement in detail here and provide links to any pre-existing/relevant documentation and screenshots if necessary:

Since a lot of people are interested in multifactor authentication, I think it might be useful to add some documentation on that. I think it can be achieved with a flow like this (error handling not included):
```
// call on your first authentication page after user presses create account button
await auth().createUserWithEmailAndPassword(email, password);
// call after user enters phone number and presses send verification code button
const confirmation = await auth().verifyPhoneNumber(phone);
/// call after getting verification code from user
const credential = auth.PhoneAuthProvider.credential(
      confirmation.verificationId,
      code,
 );
// call after successful code verification, to resend code call the verifyPhoneNumber function again
await auth().currentUser.linkWithCredential(credential);

```
---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

### PR DESCRIPTION
**Pull request by g4laura**
Update phone-auth to include MFA flow

### Description

A lot of apps require users to verify their phone number after creating an account with an email and password, this documentation would explain how to do that using the verifyPhoneNumber method.

### Related issues
Fixes #4418 

### Release Summary

Updates documentation to include a MFA flow, using auth().verifyPhoneNumber method to link a phone number with the currently authenticated user.

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [ x] Yes
- My change supports the following platforms;
  - [x ] `Android`
  - [x ] `iOS`
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [ x ] No

### Test Plan

Example of firebase user verified using this method

Screen Shot 2020-10-20 at 7 12 04 PM
---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

:fire:

### CODE REVIEW

File: docs/auth/phone-auth.md
Comment (mikehardy) [inline docs/auth/phone-auth.md:131]:
```suggestion
        onPress={() => verifyPhoneNumber('ENTER A VALID TESTING OR REAL PHONE NUMBER HERE')}
```

File: docs/auth/phone-auth.md
Comment (mikehardy) [inline docs/auth/phone-auth.md:90]: Could you add something here that indicates firebase is not a standalone component in an MFA system? It appears to require enrollment in the Google Cloud Identify Platform as well? Anything similar to the answer here might serve well: https://stackoverflow.com/a/60719962/9910298

**Review by mikehardy — CHANGES_REQUESTED** [review]
Hi there!
I think this is close but without a mention that there is more work to do for MFA to work I am certain it will increase support load in the issue tracker here, and that's a precious resource. Anything that can be done that points potential implementors to the official upstream documentation that fully describes MFA would be welcome

File: docs/auth/phone-auth.md
Comment (g4laura) [reply docs/auth/phone-auth.md:90]: How is this?

**Review by mikehardy — APPROVED** [review]
This looks fantastic, thanks @g4laura !

**Comment by g4laura** [comment]
> This looks fantastic, thanks @g4laura !

Thank you! 🙂

**Comment by leonid-shevtsov** [comment]
Hello, `linkWithCredential` does not set up MFA - it adds a _second login method_ to the account. So it allows the user to login with _either_ email&password _or_ sms to phone number. So it actually **makes the account less secure**. Suggesting people use it as a MFA replacement is dangerously misleading.

MFA requires a separate API - in the Web JS SDK it's exposed at `user.multiFactor`: https://firebase.google.com/docs/reference/js/firebase.User#multifactor

**Comment by mikehardy** [comment]
In the implementation you may see where the credential came from and enforce that both methods are used prior to allowing access, or that specific parts of the app are done only after using the second method. I don't disagree it can be misused, but suggesting anything security related is dangerously misleading without knowing the full context of their implementation is perhaps dangerously misleading?

That said, if you have some clarifying statement you'd propose please make a PR to the docs, and if you'd like to make a PR to expose the real MFA APIs that exist on the SDKs, they would be most welcome

**Comment by hhemmati81** [comment]
The auth package from React Native Firebase is missing the resolver. Without it, you can't implement MFA that Google recently added to Firebase.

The login error code must return `auth/multi-factor-auth-required` and pass a resolver in the login error. I am not even sure how you would be able to do the recaptchaVerifier in React Native.

**Comment by mikehardy** [comment]
@hhemmati81 if this is of interest to you at all, as current maintainer here I collaborate with anyone that proposes a PR and we merge about 100% of them. The tests directory here has a readme showing how to get the test app running, and developing for this module is actually pretty efficient. I'd be happy to merge a PR that allowed 2FA if you proposed one

**Pull request merged**
